packages/store-sync/src/sqlite/sqliteStorage.ts

```diff
@@ -0,0 +1,186 @@
+                eq(mudStoreTables.address, address),
+                eq(mudStoreTables.namespace, namespace),
+                eq(mudStoreTables.name, name)
+              )
+            )
+            .run();
+        }
+
+        for (const operation of operations) {
+          const table = tables.find(
+            (table) =>
+              table.address === getAddress(operation.log.address) &&
+              table.namespace === operation.namespace &&
+              table.name === operation.name
+          );
+          if (!table) {
+            debug(`table ${operation.namespace}:${operation.name} not found, skipping operation`, operation);
+            continue;
+          }
+
+          const sqliteTable = createSqliteTable(table);
+          const key = encodePacked(
+            operation.log.args.key.map(() => "bytes32"),
+            operation.log.args.key as Hex[]
+          );
+
+          if (operation.type === "SetRecord") {
+            debug("SetRecord", operation);
+            tx.insert(sqliteTable)
+              .values({
+                __key: key,
+                __lastUpdatedBlockNumber: blockNumber,
+                __isDeleted: false,
+                ...operation.key,
+                ...operation.value,
+              })
+              .onConflictDoUpdate({
+                target: sqliteTable.__key,
+                set: {
+                  __lastUpdatedBlockNumber: blockNumber,
+                  __isDeleted: false,
+                  ...operation.value,
+                },
+              })
+              .run();
+          } else if (operation.type === "SetField") {
+            debug("SetField", operation);
+            tx.insert(sqliteTable)
+              .values({
+                __key: key,
+                __lastUpdatedBlockNumber: blockNumber,
+                __isDeleted: false,
+                ...operation.key,
+                ...schemaToDefaults(table.valueSchema),
+                [operation.fieldName]: operation.fieldValue,
+              })
+              .onConflictDoUpdate({
+                target: sqliteTable.__key,
+                set: {
+                  __lastUpdatedBlockNumber: blockNumber,
+                  __isDeleted: false,
+                  [operation.fieldName]: operation.fieldValue,
+                },
+              })
+              .run();
+          } else if (operation.type === "DeleteRecord") {
+            // TODO: should we upsert so we at least have a DB record of when a thing was created/deleted within the same block?
+            debug("DeleteRecord", operation);
+            tx.update(sqliteTable)
+              .set({
+                __lastUpdatedBlockNumber: blockNumber,
+                __isDeleted: true,
+              })
+              .where(eq(sqliteTable.__key, key))
+              .run();
+          }
+        }
+
+        tx.insert(chainState)
+          .values({
+            schemaVersion,
+            chainId: publicClient.chain.id,
```

Comment:
it seems like the only reason we need a viem `publicClient` in here is for the chain id - would it be sufficient to just pass in the chain id?

Reply:
gotcha, makes sense